src/main/java/VASSAL/configure/ConfigureTree.java

```diff
@@ -284,6 +321,205 @@ protected JPopupMenu buildPopupMenu(final Configurable target) {
+  public final int getNumMatches(String searchString) {    
+    List<DefaultMutableTreeNode> searchNodes = getSearchNodes((DefaultMutableTreeNode)getModel().getRoot());
+    int hits = 0;
+
+    for (int index = 0; index < searchNodes.size(); index++) {
+      if (checkNode(searchNodes.get(index), searchString)) {
+        hits++;
+      }
+    }
+    
+    return hits;    
+  }
+  
+    
+  // Search action - runs search dialog box, then searches
+  protected Action buildSearchAction(final Configurable target) {
+    final Action a = new AbstractAction(searchCmd) {
+      private static final long serialVersionUID = 1L;
+
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        
+        final JDialog d = new JDialog((Frame) SwingUtilities.getAncestorOfClass(Frame.class, ConfigureTree.this), true);
+        d.setTitle(searchCmd);
+        d.setLayout(new BoxLayout(d.getContentPane(), BoxLayout.Y_AXIS));
+        Box box = Box.createHorizontalBox();
+        box.add(new JLabel("String to find: "));
+        box.add(Box.createHorizontalStrut(10));
+        
+        final JTextField search = new JTextField(searchString, 32);
+        box.add(search);
+        
+        d.add(box);
+
+        box = Box.createHorizontalBox();
+        final JCheckBox sensitive = new JCheckBox(Resources.getString("Editor.search_case"), matchCase);
+        box.add(sensitive);
+        final JCheckBox names = new JCheckBox(Resources.getString("Editor.search_names"), matchNames);
+        box.add(names);        
+        final JCheckBox types = new JCheckBox(Resources.getString("Editor.search_types"), matchTypes);
+        box.add(types);        
+        d.add(box);
+        
+        box = Box.createHorizontalBox();     
+        JButton find = new JButton(Resources.getString("Editor.search_next"));
+        find.addActionListener(new ActionListener() {
+          @Override
+          public void actionPerformed(ActionEvent e) {
+            boolean anyChanges = (!searchString.equals(search.getText())) || 
+                                 (matchCase != sensitive.isSelected()) ||
+                                 (matchNames != names.isSelected()) ||
+                                 (matchTypes != types.isSelected());
+            
+            searchString = search.getText();
+            matchCase    = sensitive.isSelected();
+            matchNames   = names.isSelected();
+            matchTypes   = types.isSelected();
+            
+            if (!matchNames && !matchTypes) {              
+              matchNames = true;
+              names.setSelected(true);
+              chat (Resources.getString("Editor.search_all_off"));                            
+            }
+            
+            if (!searchString.isEmpty()) {
+              if (anyChanges) {
+                int matches = getNumMatches(searchString);
+                chat (String.valueOf(matches) + " " + Resources.getString("Editor.search_count") + searchString);
```

Comment:
Can omit the `String.valueOf()` here and just start the concatenation with the int